### DISCUSSION TITLE
Issue upgrading from 2.9.9 to latest.

### DISCUSSION THREAD

**Discussion by dwildcash** [netbox-community/netbox#5490] [Q&A] [answered]
### Environment
* Python version: 3.6.8
* NetBox version: 2.9.9

Hi,
We are running version 2.9.9.

I ran git pull to get latest version.
than ran ./uprade.sh
I got this error:
![image](https://user-images.githubusercontent.com/36945045/102543497-625aaa80-4081-11eb-969f-6ae84fe10154.png)

I did few upgrade in the past but this error just begin to happen (it happen with the database upgrade scripts I think) Look like the comman receive 3 arguments instead of 2.

**Comment by jeremystretch** [comment]
It looks like something may have gone wrong with your upgrade. 13a13f3943c933fc43be0e135a6dc78f460fba0f recently changed how plugin validation works for the v2.10 release (see #5446), however that doesn't explain why the `validate()` signature would be different. What does `git status` show? And what plugins do you have enabled?

**Comment by dwildcash** [comment]
Git Log
![image](https://user-images.githubusercontent.com/36945045/102546137-2fb2b100-4085-11eb-90e6-731b418b2d7f.png)

Git Status
![image](https://user-images.githubusercontent.com/36945045/102546187-40fbbd80-4085-11eb-8a72-859a0843f286.png)

Plugin
django3_saml2_nbplugin

It work fine if I comment out the plugin.

**Comment by jeremystretch** [comment, marked as answer]
Ah, yeah here's the problem: The plugin uses a custom `validate()` method that's not compatible with NetBox v2.10: https://github.com/jeremyschulman/netbox-plugin-auth-saml2/blob/79d76511ce2349aac4803a398d7cd1d14048f5a7/django3_saml2_nbplugin/__init__.py#L27

You can modify the plugin's config class locally to fix it, but you'll also want to open a bug report for the plugin.

**Comment by dwildcash** [comment]
Thanks for the quick answer jeremy.

Can you tell me what I need to change in file locally exactly?  I dont know exactly...

**Comment by jeremystretch** [comment]
In the plugin's `__init__.py` file, change

```py
    @classmethod
    def validate(cls, user_config):
        super().validate(user_config)
        SAML2_AUTH_CONFIG.update(user_config)
```

to

```python
    @classmethod
    def validate(cls, user_config, version):
        super().validate(user_config, version)
        SAML2_AUTH_CONFIG.update(user_config)
```

**Reply by dwildcash** [reply]
Just to let you know its fixed.  I dont know if its related to this release but I had to chmod 755 all my static content because css were not loading...  Thanks again for you help.  I will not spam anymore this thread sorry

**Reply by ylmrx** [reply]
Unless you have a plugin trying to issue "chmod" commands, it's most-likely unrelated.

**Comment by DanSheps** [comment]
Converted this to a discussion as this isn't an issue with netbox core

**Reply by chicks-net** [reply]
Thanks.  It is great that this is possible now. 😄

**Comment by ylmrx** [comment]
The saml2 plugin was updated today and solves this issue :100:  , just : 
`pip install --upgrade netbox-plugin-auth-saml2`
